File: BAEKJOON/java/14912_숫자_빈도수.java
import java.io.*;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;

public class Main {
    public static int n;
    public static char digit;
    public static Map<Character, Integer> map = new HashMap<>();

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        digit = st.nextToken().charAt(0);
    }

    private static void solution() throws IOException {
        for (int i = 1; i <= n; i++) {
            String s = String.valueOf(i);

            for (int j = 0; j < s.length(); j++) {
                map.put(s.charAt(j), map.getOrDefault(s.charAt(j), 0) + 1);
            }
        }

        bw.write(String.valueOf(map.get(digit)));
    }
}

File: BAEKJOON/cpp/14912_숫자_빈도수.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

int n;
char digit;
unordered_map<char, int> um;

void input() {
	cin >> n >> digit;
}

void solution() {
	for (int i = 1; i <= n; i++) {
		string s = to_string(i);

		for (int j = 0; j < s.size(); j++) {
			um[s[j]]++;
		}
	}

	cout << um[digit];
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}